xfid.go

```diff
@@ -925,46 +925,37 @@ func xfidutfread(x *Xfid, t *Text, q1 int, qid int) {
 func xfidruneread(x *Xfid, t *Text, q0 int, q1 int) int {
 	// log.Println("xfidruneread", x)
 	// defer log.Println("done xfidruneread")
-	var (
-		fc plan9.Fcall
-		w  *Window
-	)
 
-	w = t.w
-	w.Commit(t)
+	t.w.Commit(t)
+
 	// Get Count runes, but that might be larger than Count bytes
 	nr := min(q1-q0, int(x.fcall.Count))
 	tmp := make([]rune, nr)
 	t.file.b.Read(q0, tmp)
 	buf := []byte(string(tmp))
-	// Now chop, and back up the end until we have a full rune
-	buf = buf[:nr]
-	i := nr - utf8.UTFMax
-	// Find a full rune to start in the last 4 bytes
-	for len(buf[i:]) > 0 {
-		ru, count := utf8.DecodeRune(buf[i:])
-		if ru == utf8.RuneError {
-			i += count
-		} else {
-			break
-		}
-	}
-	// add all further full runes
-	for len(buf[i:]) > 0 {
-		ru, count := utf8.DecodeRune(buf[i:])
-		if ru == utf8.RuneError {
-			break
-		} else {
-			i += count
+
+	m := len(buf)
+	if len(buf) > int(x.fcall.Count) {
```

Comment:
Per above: conceivably I'm not understanding this code. But,  it seems like our goal is to end up with  `buf` that contains at most `x.fcall.Count` bytes so long as `buf` is a complete valid UTF8 byte sequence -- in particular no partial rune encodings?

Reply:
That's exactly right. I'll add a method doc.